src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/common/PersistenceService.java

```diff
@@ -245,23 +249,96 @@ public DatabaseChangelogLockEntity releaseDatabaseLock(int id) {
+                .and(searchByReadOnly(queryRequests.getQueryRequests().get(0).isReadOnly()));
+
+
+            if (queryRequests.getQueryRequests().size() > 1) {
+                for (int i = 1; i < queryRequests.getQueryRequests().size(); i++) {
+                    finalQuery = initialQuery.or(searchByRoleName(queryRequests.getQueryRequests().get(i)
+                                                                      .getRoleName())
+                           .and(searchByHasAttributes(queryRequests.getQueryRequests().get(i).getHasAttributes()))
+                           .and(searchByAuthorisations(queryRequests.getQueryRequests().get(i).getAuthorisations()))
+                           .and(searchByActorIds(queryRequests.getQueryRequests().get(i).getActorId()))
+                           .and(searchByGrantType(queryRequests.getQueryRequests().get(i).getGrantType()))
+                           .and(searchByValidDate(queryRequests.getQueryRequests().get(i).getValidAt()))
+                           .and(searchByAttributes(queryRequests.getQueryRequests().get(i).getAttributes()))
+                           .and(searchByRoleType(queryRequests.getQueryRequests().get(i).getRoleType()))
+                           .and(searchByClassification(queryRequests.getQueryRequests().get(i).getClassification()))
+                           .and(searchByRoleCategories(queryRequests.getQueryRequests().get(i).getRoleCategory()))
+                           .and(searchByReadOnly(queryRequests.getQueryRequests().get(i).isReadOnly())));
+
+                }
+            } else {
+                finalQuery = initialQuery;
+            }
+
+        }
+
+
+        pageRoleAssignmentEntities = roleAssignmentRepository.findAll(
+            finalQuery,
+            PageRequest.of(
+                (pageNumber != null
+                    && pageNumber > 0) ? pageNumber : 0,
+                (size != null
+                    && size > 0) ? size : defaultSize,
+                Sort.by(
+                    (direction != null) ? Sort.Direction.fromString(direction) : Sort.DEFAULT_DIRECTION,
+                    (sort != null) ? sort : sortColumn
+                )
+            )
+        );
+
+        return prepareQueryRequestResponse(existingFlag, startTime);
+    }
+
+    private List<Assignment> prepareQueryRequestResponse(boolean existingFlag, long startTime) {
+        List<Assignment> roleAssignmentList;
         if (!existingFlag) {
             roleAssignmentList = pageRoleAssignmentEntities.stream()
                 .map(role -> persistenceUtil.convertEntityToRoleAssignment(role))
                 .collect(Collectors.toList());
 
+
         } else {
             roleAssignmentList = pageRoleAssignmentEntities.stream()
                 .map(role -> persistenceUtil.convertEntityToExistingRoleAssignment(role))
                 .collect(Collectors.toList());
 
+
```

Comment:
I observed a bug in the perftest environment due to this logic. As we are natively making use of same query end point for retrieving the existing records (with some default page params)..If an actor is having say 30 records and default page size is set to 20 then only 20 existing records will be injected in kie session and there is high possibility of missing the legit existing role which can eventually reject the create or delete operation...We need to find more robust approach to handle this scenario here as we don't know the limit on the number of roles for an actor.

Reply:
Not getting.